### PR TITLE
DSND-779: added required changes for invoking chs-kafka-api before de…

### PR DESCRIPTION
**Pull request by vtondepu**
…leting charge

DSND-779: Added all required changes for invoking chs-kafka-api before deleting charges from Mortgages.
Added/modified ChargesService, ChargesAPiService and Unit tests.

### CODE REVIEW

**Comment by vtondepu** [comment]
DSND-779: Added all required changes for invoking chs-kafka-api before deleting charges from Mortgages.
Added/modified ChargesService, ChargesAPiService and Unit tests.

File: src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java [hunk 1]
```diff
@@ -61,19 +67,68 @@ public ApiResponse<Void> invokeChsKafkaApi(String contextId, String companyNumbe
...
         changedResource.setResourceKind(resourceKind);
         changedResource.setContextId(contextId);
 
         return changedResource;
     }
 
+    private PrivateChangedResourcePost getChangedResourcePost(String uri,
+                                                              ChangedResource changedResource) {
+
+        InternalApiClient internalApiClient = apiClientServiceImpl.getInternalApiClient();
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java:92]: Nice to have: Maybe create InternalApiClient Spring bean like we did in charges-delta-consumer and autowire instead?

File: src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java [hunk 2]
```diff
@@ -204,23 +208,45 @@ private void saveAndInvokeChsKafkaApi(String contextId, String companyNumber,
...
+    public void deleteCharge(String companyNumber,
+                             String contextId,
+                             String chargeId) {
         try {
-            if (!chargesRepository.existsById(chargeId)) {
+            Optional<ChargesDocument> chargesDocumentOptional =
+                    chargesRepository.findById(chargeId);
+
+            if (chargesDocumentOptional.isEmpty()) {
                 throw new IllegalArgumentException(String.format(
```
Comment (zahidm3) [inline src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java:223]: If charges document does not exists, it should be 400 bad request response back, not sure if this exception results in 400?

Comment (vtondepu) [reply src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java:223]: I will check the type of exception and set the appropriate code in controller. I think all the exceptions stuff will be streamlined part of a new story which is yet to be worked DSND-781. All status code related exceptions will be covered in this story.

File: src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java (hunk 1 above)
Comment (vtondepu) [reply src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java:92]: This is an existing code part - i just moved it to a new method to facilitate and reuse it for the delete functionality. if you want this to be improvised i can do it part of unhappy path http status codes story (DSND-781).

Comment (zahidm3) [reply src/main/java/uk/gov/companieshouse/charges/data/api/ChargesApiService.java:92]: sure, lets do it in 781

File: src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java (hunk 2 above)
Comment (zahidm3) [reply src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java:223]: sure, lets do in 781

**Review by zahidm3 — APPROVED** [review]

**Pull request merged**